_observing-your-data/forecast/security.md

```diff
@@ -0,0 +1,462 @@
+2. **Result isolation** – Document-level security (DLS) limits access to forecast results in `opensearch-forecast-result*`.
+3. **Source data access** – A minimal read-only role enables each forecaster to scan its own index.
+
+The following sections explain how to configure each layer.
+
+### Assign backend roles to users
+
+In most environments, backend roles are assigned through LDAP or SAML. However, if you are using the internal user database, you can set them manually as shown in the following example:
+
+```json
+# Analyst
+PUT _plugins/_security/api/internalusers/alice
+{
+  "password": "alice",
+  "backend_roles": ["analyst"]
+}
+
+# HR staff
+PUT _plugins/_security/api/internalusers/bob
+{
+  "password": "bob",
+  "backend_roles": ["human-resources"]
+}
+```
+
+These backend roles can then be used to control access to forecasters and forecast results on a per-team basis.
+
+### Enable backend-role filtering for configuration access
+
+To isolate forecaster configurations by team, enable backend-role filtering at the cluster level:
+
+
+```bash
+PUT _cluster/settings
+{
+  "persistent": {
+    "plugins.forecast.filter_by_backend_roles": true
+  }
+}
+```
+
+When this setting is enabled, OpenSearch records the creator’s backend roles in each forecaster document. Only users with a matching backend role can view, edit, or delete that forecaster.
+
+### Create a result‑access role per team
+
+Forecast results are stored in shared indexes, so use document-level security (DLS) to restrict access by backend role.
+
+The following example creates a role that allows users with the `analyst` backend role to read and write only their team’s forecast results:
+
+
+```json
+PUT _plugins/_security/api/roles/forecast_analyst_result_access
+{
+  "index_permissions": [{
+    "index_patterns": ["opensearch-forecast-result*"],
+    "dls": """
+    {
+      "bool": {
+        "filter": [{
+          "nested": {
+            "path": "user",
+            "query": {
+              "term": {
+                "user.backend_roles.keyword": "analyst"
+              }
+            },
+            "score_mode": "none"
+          }
+        }]
+      }
+    }""",
+    "allowed_actions": ["read","write"]
+  }]
+}
+```
+
+To isolate results for another team, such as `human-resources`, create a separate role (for example, `forecast_human_resources_result_access`) and update the term value to match the appropriate backend role.
+
+### Define data-source read access
+
+The `data_source_read` role is defined the same way as in earlier examples. It grants minimal read access to the metrics index that each forecaster uses for training and prediction.
+
+You can reuse this role across teams, or create separate versions if you need per-index restrictions.
+
+### Map a user to three roles
+
+The following example maps the user `alice` to all three required roles—forecaster control, result access, and source index read—using the `analyst` backend role:
```

Comment:
This is getting confusing. Should we use the actual names of the roles here (in code font)?